src/jit-diff/jit-diff.cs

```diff
@@ -159,25 +184,240 @@ private void SetRID()
                 switch (_platformName)
                 {
                     case "Windows":
-                    case "Linux" :
+                    case "Linux":
+                        {
+                            _moniker = _platformName;
+                        }
+                        break;
+                    case "Darwin":
+                        {
+                            _moniker = "OSX";
+                        }
+                        break;
+                    default:
+                        {
+                            Console.WriteLine("No platform mapping!");
+                            _moniker = "bogus";
+                        }
+                        break;
+                }
+            }
+
+            private string GetBuildOS()
+            {
+                switch (_platformName)
+                {
+                    case "Windows":
+                        return "Windows_NT";
+                    case "Linux":
+                        return "Linux";
+                    case "Darwin":
+                        return "OSX";
+                    default:
+                        Console.Error.WriteLine("No platform mapping!");
+                        return null;
+                }
+            }
+
+            private void SetDefaults()
+            {
+                // If --diff_root wasn't specified, see if we can figure it out from the current directory
+                // using git. NOTE: we shouldn't do this unless it will be used, that is, if there are some
+                // arguments that are currently unspecified and need this to compute their default.
+                if (_diffRoot == null)
+                {
+                    _diffRoot = GetRepoRoot();
+                }
+
+                if ((_outputPath == null) && (_diffRoot != null))
+                {
+                    _outputPath = Path.Combine(_diffRoot, s_defaultDiffDirectoryName);
+                    PathUtility.EnsureDirectoryExists(_outputPath);
+
+                    Console.WriteLine("Using --output={0}", _outputPath);
+                }
+
+                bool needTestTree = Benchmarks || DoTestTree;
+
+                if ((_platformPath == null) || (_basePath == null) || (_diffPath == null) || ((_testPath == null) && needTestTree))
+                {
+                    // Try all combinations of build and architecture to find one that has both a Core_Root
+                    // and, if necessary, base and diff builds and tests.
```

Comment:
I'm not sure I agree. I never use release. And the tool does tell you when it uses a default, what it is.